includes/class-freemius.php

```diff
@@ -7544,6 +7545,45 @@ function _activate_plugin_event_hook() {
             $this->_storage->was_plugin_loaded = true;
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.8.0
+         *
+         * @param string $plugin
+         * @param bool   $network_wide
+         */
+        function _activated_plugin( $plugin, $network_wide ) {
+            // We cannot rely on fs_remove_sdk_reference_by_basename() for reordering, as the latest SDK reference might be associated with a different product, leading to the reordering of that product instead of the intended one.
+            $this->move_free_plugin_to_end_of_active_plugins( $this->_free_plugin_basename );
+
+            if ( $plugin !== $this->_free_plugin_basename ) {
+                /**
+                 * To avoid placing the free version at the start of the active plugins option when the SDK is loaded from it, remove the SDK reference linked to the free version if it is the newest.
+                 */
+                fs_remove_sdk_reference_by_basename( $this->_free_plugin_basename );

```

Comment:
Will it hurt us even if the SDK reference from the free version is at top? I don't mind doing this "just in case", but just asking for my understanding.